### PR TITLE
OHRM-97: Security Enhancements

### PR DESCRIPTION
**Pull request by RajithaKumara**

### CODE REVIEW

File: symfony/plugins/orangehrmCorePlugin/modules/core/actions/ohrmBaseAction.class.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * OrangeHRM base action class
+ */
+abstract class ohrmBaseAction extends sfAction
+{
+
+    /**
+     * Handle response code as a bad request
+     */
+    protected function handleBadRequest()
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmCorePlugin/modules/core/actions/ohrmBaseAction.class.php]: change it to final so subclass cant override 
```
protected final function
```

File: symfony/plugins/orangehrmCorePlugin/modules/core/actions/ohrmBaseAction.class.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        $response = $this->getResponse();
+        $response->setStatusCode(HttpResponseCode::HTTP_BAD_REQUEST);
+    }
+
+    /**
+     * Forward to defined secure action in the secure module
+     * @throws sfStopException
+     */
+    protected function forwardToSecureAction()
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmCorePlugin/modules/core/actions/ohrmBaseAction.class.php]: change it to final so subclass cant override 
```
protected final function
```

File: symfony/plugins/orangehrmCorePlugin/modules/core/actions/ohrmBaseAction.class.php (hunk 1 above)
Comment (RajithaKumara) [reply symfony/plugins/orangehrmCorePlugin/modules/core/actions/ohrmBaseAction.class.php]: Done

File: symfony/plugins/orangehrmCorePlugin/modules/core/actions/ohrmBaseAction.class.php (hunk 2 above)
Comment (RajithaKumara) [reply symfony/plugins/orangehrmCorePlugin/modules/core/actions/ohrmBaseAction.class.php]: Done

**Pull request merged**